syntax = "proto3";

package NullSpaceIPC;



message DeviceConnected {
	string product_name  = 2;
	uint32 firmware_major = 3;
	uint32 firmware_minor = 4;
	uint32 firmware_patch = 5;
}


message DeviceDisconnected {
	enum DisconnectionReason {
		UNKNOWN = 0;
		UNPLUGGED = 1;
		CRASH = 2;
	}

	DisconnectionReason reason = 1;
}


message DeviceEvent {
	uint32 device_id = 1;

	oneof events {
		DeviceConnected connected_event = 2;
		DeviceDisconnected disconnected_event = 3;
	}
}
